string_utils/fms_string_utils.F90

```diff
@@ -0,0 +1,148 @@
+!***********************************************************************
+!*                   GNU Lesser General Public License
+!*
+!* This file is part of the GFDL Flexible Modeling System (FMS).
+!*
+!* FMS is free software: you can redistribute it and/or modify it under
+!* the terms of the GNU Lesser General Public License as published by
+!* the Free Software Foundation, either version 3 of the License, or (at
+!* your option) any later version.
+!*
+!* FMS is distributed in the hope that it will be useful, but WITHOUT
+!* ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+!* FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+!* for more details.
+!*
+!* You should have received a copy of the GNU Lesser General Public
+!* License along with FMS.  If not, see <http://www.gnu.org/licenses/>.
+!***********************************************************************
+
+!> @defgroup fms_string_utils_mod fms_string_utils_mod
+!> @ingroup string_utils
+!> @brief Routines to use for string manipulation
+
+!> @file
+!> @brief File for @ref fms_string_utils_mod
+
+!> @addtogroup fms_string_utils_mod
+!> @{
+module fms_string_utils_mod
+  use, intrinsic :: iso_c_binding
+  use fms_mod, only: fms_c2f_string
```

Comment:
Can you move `fms_c2f_string` here, then have fms_mod `use fms_string_utils_mod, only:fms_c2f_string`.  I think you want the dependency going the other way instead...